compendium/DeclarativeServices/src/manager/ComponentManagerImpl.cpp

```diff
@@ -158,25 +211,64 @@ namespace cppmicroservices
 
         std::shared_future<void>
         ComponentManagerImpl::PostAsyncDisabledToEnabled(
-            std::shared_ptr<cppmicroservices::scrimpl::ComponentManagerState>& currentState)
+            std::shared_ptr<cppmicroservices::scrimpl::ComponentManagerState>& currentState,
+            std::shared_ptr<std::atomic<bool>> asyncStarted)
         {
             auto metadata = GetMetadata();
             auto bundle = GetBundle();
             auto reg = GetRegistry();
             auto logger = GetLogger();
             auto configNotifier = GetConfigNotifier();
 
-            using ActualTask = std::packaged_task<void(std::shared_ptr<CMEnabledState>)>;
+            using ActualTask = std::packaged_task<void(std::shared_ptr<CMEnabledState>, bool)>;
             using PostTask = std::packaged_task<void()>;
 
-            ActualTask task([metadata, bundle, reg, logger, configNotifier](
-                                std::shared_ptr<CMEnabledState> eState) mutable
-                            { eState->CreateConfigurations(metadata, bundle, reg, logger, configNotifier); });
+            ActualTask task(
+                [metadata, bundle, reg, logger, configNotifier, asyncStarted](std::shared_ptr<CMEnabledState> eState,
```

Comment:
all but asyncStarted were there before -- left alone
asyncStarted was added by me and I copied it as it's a shared_ptr. Have to hang on to the object, so copied it rather than a reference to a variable on the stack